File: pset6/readability/readability.py
from cs50 import get_string

text = get_string("Text:")
words = 1
index = sentence = letters = 0 
for i in range(len(text)):
    if text[i] in ["!", ".", "?"]:
        sentence += 1
    elif (text[i].isalpha()) == True:
        letters += 1
    elif text[i] == " ":
        words +=1
    else:
        pass

index = round(0.0588 * (letters / words * 100) - 0.296 * (sentence / words * 100) - 15.8)

if index < 1:
    print("Before Grade 1")
elif index > 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")

exit(0)    

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <ctype.h>
#include <math.h>

#define MAX_WORDS 100

int main (void){
    string text=NULL;
    char *p;
    int letters = 0, words = 1, sentence = 0, index; 
    
    text = get_string("Text:");
    p=text;
    
    while((*p=toupper(*p))){
       if (*p == '!' || *p == '.' || *p == '?')
            sentence ++;
        else if (*p >= 'A' && *p<= 'Z')
            letters ++;
        else if (*p == ' ')
            words ++;
        else 
            ;
        p++;
    }
    
    index=round(0.0588f*((float)letters/words*100)-0.296f*((float)sentence/words*100)-15.8);
    
    if(index<1)
        printf("Before Grade 1\n");
    else if(index>16)
        printf("Grade 16+\n");
    else 
        printf("Grade %d\n", index);
        
    return 0;
}
